astropy/io/misc/yaml.py

```diff
@@ -328,15 +329,30 @@ def represent_float(self, data):
     AstropyDumper.add_multi_representer(cls, _quantity_representer(tag))
     AstropyLoader.add_constructor(tag, _quantity_constructor(cls))
 
-for cls in list(coords.representation.REPRESENTATION_CLASSES.values()) + list(
-    coords.representation.DIFFERENTIAL_CLASSES.values()
-):
+# Add representations, differentials, frames defined in astropy.
+cls_coords = []
+cls_coords.extend(
+    cls
+    for cls in itertools.chain(
+        coords.representation.REPRESENTATION_CLASSES.values(),
+        coords.representation.DIFFERENTIAL_CLASSES.values(),
+    )
+    if cls.__name__ in coords.representation.__all__
```

Comment:
How about here and below using `if cls.name in coords.__dict__`? (or `... in dir(coords)`). Since we `"astropy.coordinates." + name` below, that would avoid surprises (though right now these are, I think, identical).